### PR TITLE
CAP-26 Implements CRUD company

### PR DESCRIPTION
**Pull request by CaoSon1909**

### CODE REVIEW

File: src/main/java/org/capstone/job_fair/controllers/payload/CreateCompanyResponse.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.capstone.job_fair.controllers.payload;
+
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/controllers/payload/CreateCompanyResponse.java:2]: Why do we need this class, we already have GenericMessageResponseEntity.

File: src/main/java/org/capstone/job_fair/config/SecurityConfig.java [hunk 2]
```diff
@@ -81,7 +81,9 @@ protected void configure(HttpSecurity http) throws Exception {
                 .authorizeRequests().antMatchers(ApiEndPoint.Account.REGISTER_COMPANY_MANAGER).permitAll().and()
                 //ResetPassword API Security
                 .authorizeRequests().antMatchers(ApiEndPoint.Authentication.REFRESH_TOKEN_ENDPOINT).permitAll().and()
-                .authorizeRequests().antMatchers(ApiEndPoint.Authentication.GENERATE_OTP_ENDPOINT).permitAll()
+                .authorizeRequests().antMatchers(ApiEndPoint.Authentication.GENERATE_OTP_ENDPOINT).permitAll().and()
+                //Company API Security
+                .authorizeRequests().antMatchers(ApiEndPoint.Company.COMPANY_ENDPOINT).permitAll()
                 .anyRequest().authenticated().and()
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/config/SecurityConfig.java:87]: Why company endpoint get permit all

File: src/main/java/org/capstone/job_fair/controllers/company/CompanyController.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    private static final String DELETE_COMPANY_SUCCESS = "Delete company sucessfully.";
+    private static final String DELETE_COMPANY_FAIL= "Delete company failed.";
+    private static final String SIZE_INVALID= "Chosen size is invalid";
+
+    @Autowired
+    private CompanyService companyService;
+
+    @Autowired
+    private CompanySizeRepository sizeRepository;
+
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/controllers/company/CompanyController.java:36]: Controller should never have Repository, please use Service

File: src/main/java/org/capstone/job_fair/controllers/company/CompanyController.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @DeleteMapping(ApiEndPoint.Company.COMPANY_ENDPOINT + "/"+"{id}")
+    public ResponseEntity<?> delete(@PathVariable String id) {
+        Boolean result = companyService.deleteCompany(id);
+        return result ? new ResponseEntity<>(new UpdateCompanyResponse(DELETE_COMPANY_SUCCESS), HttpStatus.OK)
+                : new ResponseEntity<>(new UpdateCompanyResponse(DELETE_COMPANY_FAIL), HttpStatus.BAD_REQUEST);
+    }
+
+    private CompanyDTO mappingRequestToDTO(CreateCompanyRequest request) {
+        CompanyDTO dto = new CompanyDTO();
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/controllers/company/CompanyController.java:95]: Please check Bảo's code for builder pattern

**Review by TienTruong2000 — CHANGES_REQUESTED** [review]

File: src/main/java/org/capstone/job_fair/services/interfaces/company/CompanyService.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+import org.capstone.job_fair.models.dtos.company.CompanyDTO;
+import org.capstone.job_fair.models.entities.company.CompanyEntity;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface CompanyService {
+
+    List<CompanyEntity> getAllCompanies();
```
[the remark this answers is not in the document]
Comment (CaoSon1909) [reply src/main/java/org/capstone/job_fair/services/interfaces/company/CompanyService.java:11]: findAll() trả về List<CompanyEntity> mà

**Review by TienTruong2000 — APPROVED** [review]
LGTM

**Pull request merged**